CIScripts/Tests/AgentTests.ps1

```diff
@@ -1,12 +1,33 @@
+$DefineTestIfAnOutputSuggestsThatAllTestsHavePassed = {
+    function Test-IfAnOutputSuggestsThatAllTestsHavePassed {
+        Param ([Parameter(Mandatory = $true)] [Object[]] $TestOutput)
+        $NumberOfTests = -1
+        Foreach ($Line in $TestOutput) { #$Lines) {
+            if ($Line -match "\[==========\] (?<HowManyTests>[\d]+) test[s]? from [\d]+ test [\w]* ran[.]*") {
+                $NumberOfTests = $matches.HowManyTests
+            }
+            if ($Line -match "\[  PASSED  \] (?<HowManyTestsHavePassed>[\d]+) test[.]*" -and $NumberOfTests -ge 0) {
+                return $($matches.HowManyTestsHavePassed -eq $NumberOfTests)
+            }
+        }
+        return $False
+    }
+}
+
 function Run-Test {
     Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
-           [Parameter(Mandatory = $true)] [string] $TestExecutable)
+           [Parameter(Mandatory = $true)] [String] $TestExecutable)
     Write-Host -NoNewline "===> Agent tests: running $TestExecutable... "
     $Res = Invoke-Command -Session $Session -ScriptBlock {
         $Res = Invoke-Command -ScriptBlock {
             $ErrorActionPreference = "SilentlyContinue"
-            Invoke-Expression "C:\Artifacts\$using:TestExecutable --config C:\Artifacts\vnswa_cfg.ini" | Out-Null
-            $LASTEXITCODE
+            $TestOutput = Invoke-Expression "C:\Artifacts\$using:TestExecutable --config C:\Artifacts\vnswa_cfg.ini"
+            $SeemsLegitimate = Test-IfAnOutputSuggestsThatAllTestsHavePassed -TestOutput $TestOutput
+            if($LASTEXITCODE -eq 0 -or $SeemsLegitimate) {
+                return 0
+            } else {
+                return $LASTEXITCODE
```

Comment:
Had to check if `$ErrorActionPreference` is bound to local scope. FYI it is:
```
PS C:\Users\mk> Invoke-Command -ScriptBlock {
>>    $ErrorActionPreference = "Stop"
>>    echo "1"
>>    Invoke-Command -ScriptBlock {
>>       $ErrorActionPreference = "SilentlyContinue";
>>       echo "A";
>>       throw "some error";
>>       echo "B"
>>    }
>>    throw "some other error";
>>    echo "2"
>> }
1
A
B
some other error
At line:10 char:4
+    throw "some other error";
+    ~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (some other error:String) [], RuntimeException
    + FullyQualifiedErrorId : some other error

```

Reply:
Yes, it is a workaround for changing `ErrorAction` only for command executed by `Invoke-Expression`. It is used in some more test code.